test/jit/fixtures_srcs/test_upgrader_models_generation.py

```diff
@@ -0,0 +1,20 @@
+# Owner(s): ["oncall: mobile"]
+
+import torch
+from test.jit.fixtures_srcs.generate_models import ALL_MODULES
+from torch.testing._internal.common_utils import TestCase, run_tests
+
+
+class TestUpgraderModelGeneration(TestCase):
```

Comment:
Isn't this redundant since we are basically checking the same thing as the original file.

Reply:
It's for different purposes. 

- Unit test can run per CI to protect the model generation process, so it can prevent committing changes to this process. 

- In the meanwhile, the error message in `generate_models.py` is more like, user makes changes in `generated_models.py` and the error message can provide immediate feedback about what is wrong. 

It's for addressing the feedback from @gmagogsfm in https://github.com/pytorch/pytorch/pull/72313:
> Do we have tests to verify that the key to this dictionary is a TestCase instance not a TestCase class? I think one of the common mistake is forgetting those parenthese.

Reply:
Oh got it